caom2-datalink-server/src/main/java/ca/nrc/cadc/caom2/datalink/ArtifactProcessor.java

```diff
@@ -226,6 +242,23 @@ else if (ReleaseType.META.equals(a.getReleaseType()))
                 }
             }
         }
+        log.warn("numFiles: " + numFiles);
+        if (numFiles > 1) {
+            
+            URL pkg = getBasePackageURL(uri);
+            if (pkg != null) {
+                DataLink link = new DataLink(uri.toASCIIString(), DataLink.Term.PKG);
+                try {
+                    link.url = getPackageURL(pkg, uri);
+                    link.contentType = CONTENT_TYPE_TAR;
+                    link.description = "single download containing all files (previews and thumbnails excluded)";
+                    link.readable = pkgReadable;
+                } catch (MalformedURLException ex) {
+                    link.errorMessage = "failed to create package link: " + ex;
+                }
+                ret.add(link);
+            }
+        }
```

Comment:
Do you need to add the link to ret when numFiles == 1?

Reply:
no, that link var is the pkg link in the block and we only want a pkg link with multiple files; the individual links are always added up at line ~206